target/linux/ath79/generic/base-files/etc/board.d/03_gpio_switches

```diff
@@ -73,6 +73,9 @@ tplink,wbs510-v1|\
 tplink,wbs510-v2)
 	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "20"
 	;;
+tplink,eap225-wall-v2)
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "tp-link:poe-passthrough:enable"
+	;;
```

Comment:
gpio-export and 03_gpio_switches are _alternatives_

Either keep your gpio-export implementation and drop this entry in 03_gpio_switches, or
drop gpio-export and put `ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "20"` here (so, you can effectively merge with the cpe/wbs devices).

Which one you choose is essentially up to you.

Reply:
If I only add a named export in the DTS, would that still create an entry by default in the system config? The switch is off by default, but I would like the UCI config to be present already so it is more discoverable to users.

Reply:
I'm not 100 % sure, but I think gpio-export can be toggled by the user, but not via uci:

`echo 1 > /sys/class/gpio/gpio20/value`

gpio-hog can't be toggled at all, and if you want uci config, you need to go with 03_gpio_switches.

Reply:
Going through the init code, named gpio pins in a gpio_switch config are [handled differently](https://github.com/openwrt/openwrt/blob/f1bc66c7653e7603b4621f5ea14be129a2bd5e4f/package/base-files/files/etc/init.d/gpio_switch#L46). For a numerical gpio_pin value, the gpio is exported, configured and has its value set correctly. A named gpio pin is assumed to be set up already, and only the saved value is restored. This was introduced in commit c5bf9a8ced6567c6bd9da63c6ff4c4f2f87207fd.

By using 03_gpio_switches, I can ensure /etc/config/system is preseeded with the appropriate gpio_switch section. But if you feel this is something that should be left to the user, I can drop this part and only use the DTS export.

Reply:
Since it looks like 03_gpio_switches is never used for named gpio's, I'll just stick to gpio-export. I would still like the preseeded config, but this is probably something that needs be solved in a more generic way with another patch.

Reply:
> Since it looks like 03_gpio_switches is never used for named gpio's,

Just look at how "new" the referenced commit is (and that I didn't know about it as well). But actually I'm not really sure whether I like this new setup option created there; so I'm happy with pure gpio-export as you did now.

Keep in mind that both 03_gpio_switches and gpio-export are actually just intermediate workarounds that only exist because nobody is interested in doing a proper implementation (gpio-export is naked upstream, and 03_gpio_switches is pure user-space). The available upstream solution (gpio-hog) however does not support user interaction, though ...